### PR TITLE
Add Item Form

### LINKED ISSUE #4: 4. As a user, I want to add a new item to my shopping list so I can start recording purchases.
A shopping list item consists of the following data points:

* Name of item
* How soon are you likely to buy it again?
  * Soon (in the next 7 days)
  * Kind of soon (in the next 14 days)
  * Not soon (in the next 30 days)
* Last purchased date

AC:

* User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
* When the user submits the form, the item is saved to the database, associated with the user’s token
* Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: `7` for “Soon”, `14` for “Kind of Soon”, and `30` for “Not Soon”
* The last purchased date should be set to `null` initially (i.e. you can create an item without purchasing it)
* Item names should be displayed as a simple, unordered list on the list view after they’re added
Notes:
* The add item form has the potential to be a lot of scope. The wireframe https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/add-item.png has several notes related to accessibility and UX.
* Eventually, the list will look like the following wireframe: https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png
* It's good for your team to look at the wireframe now, but be sure not to try to implement all of the visible functionality at once. There will be future stories to add all of those features.
* For this story, it's sufficient just to get a simple list of the items displaying.

### PR DESCRIPTION
**Pull request by rachael-t**
## Description

Updated the AddList UI to have a form for users to input a new item name, and select how soon they need that item.

When they click the button to "Add Item" the logic will get the user's token, and add that list item to the user's specific collection.

Note: until merged with issue #3 we needed to add imports and logic into the `AddItem.js` file to get the token and set the user so that we could get the user's token for the database. We have added comments to that file notating what we predict we will want to remove once merged with the other code.

## Related Issue

closes #4 

## Acceptance Criteria

- [x] User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
- [x] When the user submits the form, the item is saved to the database, associated with the user’s token
- [x] Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: 7 for “Soon”, 14 for “Kind of Soon”, and 30 for “Not Soon”
- [x] The last purchased date should be set to null initially (i.e. you can create an item without purchasing it)
- [x] Item names should be displayed as a simple, unordered list on the list view after they’re added

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
| ✓   | :link: Update dependencies |
|     | :scroll: Docs              |

### Before
Screen Shot 2022-04-16 at 11 54 11 AM

### After
Screen Shot 2022-04-16 at 11 53 20 AM
Screen Shot 2022-04-16 at 11 53 33 AM
Screen Shot 2022-04-16 at 11 53 41 AM

## Testing Steps / QA Criteria

- From your terminal, pull down this branch with git pull origin rt-ng-add-item-form and check that branch out with git checkout rt-ng-add-item-form 
- Then npm to install the newly added dependencies locally and npm start to launch the app.
- Click the link to "add item" at the bottom of the page. Complete the form and click on "Add Item." Then check the database to see your item added to the user's specific collection.

### CODE REVIEW

**Review by lena-A-Al — APPROVED** [review]
@rachael-t and @nikkigraybeal You amazingly put a solution for tickets 3 and 4 nicely. Your solution is well explained and logically easy to follow.

**Review by simonkaine — APPROVED** [review]
Nice work using the utility storage functions from our code base to work in a temporary solution in utilizing the user token capabilities. We've also adjusted our code to use your AddItem component logic and in a combined efforts, I believe we have successfully created a usable component that won't have any merge conflicts. Please see our latest commit ([Merged AddItem logic]) in la-sk-create-user-token branch for details.

File: src/components/AddItem.js [hunk 1]
```diff
@@ -1,24 +1,83 @@
...
+import { useState } from 'react';
+// When merging to main with Issue #3, we can remove setUser below
+import { setUser, getUser } from '../storage-utils/storage-utils';
+// When merging to main with Issue #3, we can remove the import below
+import { getToken } from '@the-collab-lab/shopping-list-utils';
 
 export default function AddItem() {
-  const submitHandler = async () => {
+  const [itemName, setItemName] = useState('');
+  const [frequency, setFrequency] = useState(7);
```
Comment (skylerwshaw) [inline src/components/AddItem.js:11]: Though not an issue for this PR, consider how we might avoid 'magic numbers' ([reference](https://codetuts.dev/magic-numbers-in-programming/)) here by creating a constant/map/object/dict/class that gives this `7` number more value and expressiveness to the code and the developer reading it!

File: src/components/AddItem.js [hunk 2]
```diff
@@ -1,24 +1,83 @@
...
+
+  const submitHandler = async (e) => {
+    e.preventDefault();
+
+    // Lines 17 - 18 are for testing purposes. Once we merge into main with Issue #3, we can remove these two lines
+    const newUserToken = getToken();
+    setUser(newUserToken);
```
Comment (skylerwshaw) [inline src/components/AddItem.js:18]: Nitpick: Let's be sure to come back and clean this up.

File: src/components/AddItem.js [hunk 3]
```diff
@@ -1,24 +1,83 @@
...
+    // This will remain:
+    const userToken = await getUser();
+    console.log('GET USER TOKEN FROM LS: ', userToken);
```
Comment (skylerwshaw) [inline src/components/AddItem.js:21]: Nitpick: We should avoid deploying console.log statements to production as they can proliferate easily. Let's be sure to clean them up.

**Review by skylerwshaw — APPROVED** [review]
Approved!

- Thank you for resetting the PR template!
- Nice form design! For later: Consider how we might simplify and make the form more dry. Could we iterate over a dictionary/object containing the frequencies and create the multiple inputs without so many lines?
- It may keep the code cleaner long-term to avoid so many log lines. They're great for debugging but less so for maintaining production code.
- These utility functions are 👩‍🍳 👌 mint.

**Pull request merged**